### PR TITLE
Check, that openapi process @Pattern annotation

### PR DESCRIPTION
**Pull request by fedinskiy**
### Summary

Openapi should add content from javax.validation.constraints.Pattern to opeapi yaml file. Implemented in https://issues.redhat.com/browse/QUARKUS-2004

Please select the relevant options.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Dependency update
- [ ] Refactoring
- [ ] Backport
- [x] New scenario (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

### Checklist:
- [x] Methods and classes used in PR scenarios are meaningful
- [x] Commits are well encapsulated and follow [the best practices](https://cbea.ms/git-commit/)

### CODE REVIEW

File: http/http-minimum/src/test/java/io/quarkus/ts/http/minimum/OpenAPIIT.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+
+import io.quarkus.test.scenarios.QuarkusScenario;
+
+@QuarkusScenario
+public class OpenAPIIT {
+    @Test
+    public void smoke() {
```
Comment (pjgg) [inline http/http-minimum/src/test/java/io/quarkus/ts/http/minimum/OpenAPIIT.java:19]: Try to find a better test name: 
Suggestion:

`smoke` -> `openapiIsReady`

File: http/http-minimum/src/test/java/io/quarkus/ts/http/minimum/OpenAPIIT.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        given().get("/api/q/openapi").then().statusCode(HttpStatus.SC_OK);
+    }
+
+    @Test
+    public void validation() throws JsonProcessingException {
```
Comment (pjgg) [inline http/http-minimum/src/test/java/io/quarkus/ts/http/minimum/OpenAPIIT.java:24]: - I would add a tag pointing to the issue that is behind this test.
- Try to find a better tets name: 
Suggestion
- `validation` -> `openapiSizeAndPatternConstraints`

**Review by pjgg — CHANGES_REQUESTED** [review]

File: http/http-minimum/src/test/java/io/quarkus/ts/http/minimum/OpenAPIIT.java (hunk 2 above)
Comment (fedinskiy) [reply http/http-minimum/src/test/java/io/quarkus/ts/http/minimum/OpenAPIIT.java]: renamed the method

**Review by pjgg — APPROVED** [review]
LGTM

**Review by rsvoboda — CHANGES_REQUESTED** [review]
`http-minimum` is getting too big, this needs to go into `http-advanced`

`-reactive` module is not enhanced

so the PR should target `http-advanced` and `http-advanced-reactive`

**Review by rsvoboda — APPROVED** [review]
Looks good now, let's wait for CI

**Comment by fedinskiy** [comment]
run tests

**Comment by pjgg** [comment]
OCP errors are not related to this PR. 
Fixed on: https://github.com/quarkus-qe/quarkus-test-suite/pull/813

**Pull request merged**